src/App.tsx

```diff
@@ -15,48 +17,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = '',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+function getSortedGoods(goodsArr: string[], sortField: SortType): string[] {
+  if (sortField !== SortType.Default) {
+    goodsArr.sort((a, b): number => {
+      switch (sortField) {
+        case SortType.Alphabet: {
+          return a.localeCompare(b);
+        }
+
+        case SortType.Length: {
+          return a.length - b.length;
+        }
+
+        default: {
+          return 0;
+        }
+      }
+    });
+  }
+
+  return goodsArr;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortField, setSortField] = useState(SortType.Default);
+  const isHideResetBtn = sortField || isReversed;
+
+  const resetField = () => {
```

Comment:
You can choose a better name for this handler:
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364